src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/InferParameterBindingInfoConvention.cs

```diff
@@ -118,13 +116,7 @@ private bool ParameterExistsInAnyRoute(ActionModel action, string parameterName)
         private bool IsComplexTypeParameter(ParameterModel parameter)
         {
             // No need for information from attributes on the parameter. Just use its type.
-            var metadata = _modelMetadataProvider
-                .GetMetadataForType(parameter.ParameterInfo.ParameterType);
-
-            if (AllowInferringBindingSourceForCollectionTypesAsFromQuery && metadata.IsCollectionType)
```

Comment:
`AllowInferringBindingSourceForCollectionTypesAsFromQuery` was one of the few properties that was `false` with recent `CompatibilityVersion`s.